src/test/java/com/google/api/generator/engine/ast/AssignmentOperationExprTest.java

```diff
@@ -0,0 +1,662 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static org.junit.Assert.assertThrows;
+
+import org.junit.Test;
+
+public class AssignmentOperationExprTest {
+  /** =========== Multiply And Assignment Operators: Variable is declaration ================ */
+  @Test
+  public void multiplyAndAssignmentOperationExpr_invalidVariableExprIsDecl() {
```

Comment:
Nit: `invalidFooOperation_reason` for consistency with the rest of this codebase, same below.

Reply:
Could we also please move the invalid cases below the valid ones (for consistency)?

Reply:
sg